donut/modules/bodfeedback/routes.py

```diff
@@ -0,0 +1,137 @@
+
+
+# api endpoint with all visible data
+@blueprint.route('/1/bodfeedback/view/<uuid:id>')
+def bodfeedback_api_view_complaint(id):
+    if not helpers.get_id(id):
+        return flask.render_template("404.html")
+    complaint_id = helpers.get_id(id)
+    #pack all the data we need into a dict
+    return flask.jsonify(helpers.get_all_fields(complaint_id))
+
+
+# view a complaint
+@blueprint.route('/bodfeedback/view/<uuid:id>')
+def bodfeedback_view_complaint(id):
+    if not helpers.get_id(id):
+        return flask.render_template("404.html")
+    complaint_id = helpers.get_id(id)
+    complaint = helpers.get_all_fields(complaint_id)
+    complaint['uuid'] = id
+    return flask.render_template('complaint.html', complaint=complaint)
+
+
+# add a message to this post
+@blueprint.route('/1/bodfeedback/add/<uuid:id>', methods=['POST'])
+def bodfeedback_add_msg(id):
+    complaint_id = helpers.get_id(id)
+    if not complaint_id:
+        flask.abort(400)
+        return
+    fields = ['message', 'poster']
+    data = {}
+    for field in fields:
+        data[field] = flask.request.form.get(field)
+    if data['message'] == "":
+        flask.abort(400)
+        return
+    helpers.add_msg(complaint_id, data['message'], data['poster'])
+    return flask.jsonify({
+        'poster': data['poster'],
+        'message': data['message']
+    })
+
+
+# allow bod members to see a summary
+@blueprint.route('/bodfeedback/view/summary')
+def bodfeedback_view_summary():
+    #get a list containing data for each post
+    complaints = helpers.get_new_posts()
+    #add links to each complaint
+    for complaint in complaints:
+        complaint['link'] = helpers.get_link(complaint['complaint_id'])
+    return flask.render_template('summary.html', complaints=complaints)
+
+
+# mark a complaint read
+@blueprint.route('/1/bodfeedback/markRead/<uuid:id>')
+def bodfeedback_mark_read(id):
+    #authenticate
+    complaint_id = helpers.get_id(id)
+    if helpers.mark_read(complaint_id) == False:
+        flask.abort(400)
+        return
+    return 'Success'
+
+
+# mark a complaint unread
+@blueprint.route('/1/bodfeedback/markUnread/<uuid:id>')
+def bodfeedback_mark_unread(id):
+    #authenticate
+
+    complaint_id = helpers.get_id(id)
+    if helpers.mark_unread(complaint_id) == False:
+        flask.abort(400)
+        return
+    return 'Success'
+
+
+# add an email to this complaint
+@blueprint.route('/1/bodfeedback/addEmail/<uuid:id>', methods=['POST'])
+def bodfeedback_add_email(id):
+    complaint_id = helpers.get_id(id)
+    if not complaint_id:
+        flask.abort(400)
+        return
+    fields = ['email']
+    data = {}
+    for field in fields:
+        data[field] = flask.request.form.get(field)
+    if data['email'] == "":
+        flask.abort(400)
+        return
+    helpers.add_email(complaint_id, data['email'])
+    return flask.jsonify({'email': data['email']})
+
+
+# remove an email from this complaint
```

Comment:
More nit: in general, comments start with a capital letter per the coding guidelines.